src/overlays/actors/ovl_En_Shn/z_en_shn.c

```diff
@@ -31,47 +65,379 @@ const ActorInit En_Shn_InitVars = {
     (ActorFunc)EnShn_Draw,
 };
 
-#endif
+static ActorAnimationEntryS sAnimations[] = {
+    { &object_shn_Anim_00D9D0, 1.0f, 0, -1, 0, 0 },
+    { &object_shn_Anim_00D9D0, 1.0f, 0, -1, 0, -4 },
+    { &object_shn_Anim_00E6C4, 1.0f, 0, -1, 0, 0 },
+    { &object_shn_Anim_00E6C4, 1.0f, 0, -1, 0, -4 },
+};
+
+static s32 D_80AE7258[] = { 0, 2, 3, 8, 10, 1 };
+
+void func_80AE6130(EnShn* this) {
+    this->skelAnime.playSpeed = this->playSpeed;
+    SkelAnime_Update(&this->skelAnime);
+}
+
+s32 func_80AE615C(EnShn* this, s32 arg1) {
+    s32 phi_v0 = 0;
+    s32 phi_v1 = 0;
+
+    if (arg1 != this->unk_2E8) {
+        phi_v0 = 1;
+    }
+    if (phi_v0 != 0) {
+        this->unk_2E8 = arg1;
+        phi_v1 = func_8013BC6C(&this->skelAnime, sAnimations, arg1);
+        this->playSpeed = this->skelAnime.playSpeed;
+    }
+    return phi_v1;
+}
+
+s32 EnShn_IsFacingPlayer(EnShn* this) {
+    s16 phi_v1;
+    f32 range;
+
+    if (this->unk_2C8 == 0) {
+        phi_v1 = 0;
+    } else {
+        this->unk_2C8--;
+        phi_v1 = this->unk_2C8;
+    }
+    if (phi_v1 == 0) {
+        this->unk_2EC ^= 1;
+        this->unk_2C8 = Rand_S16Offset(0x1E, 0x1E);
+    }
+    if (this->unk_2EC != 0) {
+        range = 120.0f;
+    } else {
+        range = 80.0f;
+    }
+    return Actor_IsFacingAndNearPlayer(&this->actor, range, 0x238C);
+}
+
+Player* EnShn_GetPlayer(EnShn* this, GlobalContext* globalCtx) {
+    return GET_PLAYER(globalCtx);
+}
+
+void func_80AE626C(EnShn* this) {
+    s32 topPad; // Vec3f Sandwhich
+    Vec3f sp44;
+    Vec3f b;
+    s32 bottomPad;
```

Comment:
```suggestion
    s32 topPad;
    Vec3f playerPos; // Used for both world and focus pos
    Vec3f thisPos; // Used for both world and focus pos
    s32 bottomPad;
```
Looks like `sp44` is used to store `player` pos data, while `b` is used to store `this` pos data.